ibc/store/provable_store_test.go

```diff
@@ -0,0 +1,540 @@
+	return utilityMock
+}
+
+func newPersistenceMock(t *testing.T,
+	bus modules.Bus,
+	dbMap map[string]string,
+) *mockModules.MockPersistenceModule {
+	t.Helper()
+
+	ctrl := gomock.NewController(t)
+	persistenceMock := mockModules.NewMockPersistenceModule(ctrl)
+	persistenceReadContextMock := mockModules.NewMockPersistenceReadContext(ctrl)
+
+	persistenceMock.EXPECT().GetModuleName().Return(modules.PersistenceModuleName).AnyTimes()
+	persistenceMock.EXPECT().Start().Return(nil).AnyTimes()
+	persistenceMock.EXPECT().Stop().Return(nil).AnyTimes()
+	persistenceMock.EXPECT().SetBus(gomock.Any()).Return().AnyTimes()
+	persistenceMock.EXPECT().GetBus().Return(bus).AnyTimes()
+	persistenceMock.EXPECT().NewReadContext(gomock.Any()).Return(persistenceReadContextMock, nil).AnyTimes()
+
+	persistenceMock.EXPECT().ReleaseWriteContext().Return(nil).AnyTimes()
+
+	persistenceReadContextMock.
+		EXPECT().
+		GetIBCStoreEntry(gomock.Any(), gomock.Any()).
+		DoAndReturn(
+			func(key []byte, _ int64) ([]byte, error) {
+				value, ok := dbMap[hex.EncodeToString(key)]
+				if !ok {
+					return nil, coreTypes.ErrIBCKeyDoesNotExist(string(key))
+				}
+				bz, err := hex.DecodeString(value)
+				if err != nil {
+					return nil, err
+				}
+				if bytes.Equal(bz, nil) {
+					return nil, coreTypes.ErrIBCKeyDoesNotExist(string(key))
+				}
+				return bz, nil
+			}).
+		AnyTimes()
+
+	persistenceReadContextMock.
+		EXPECT().
+		Release().
+		AnyTimes()
+
+	return persistenceMock
+}
+
+func newTreeStoreMock(t *testing.T,
+	bus modules.Bus,
+	tree *smt.SMT,
+	nodeStore kvstore.KVStore,
+) *mockModules.MockTreeStoreModule {
+	t.Helper()
+
+	ctrl := gomock.NewController(t)
+	treeStoreMock := mockModules.NewMockTreeStoreModule(ctrl)
+	treeStoreMock.EXPECT().GetModuleName().Return(modules.TreeStoreModuleName).AnyTimes()
+	treeStoreMock.EXPECT().SetBus(gomock.Any()).Return().AnyTimes()
+	treeStoreMock.EXPECT().GetBus().Return(bus).AnyTimes()
+
+	treeStoreMock.
+		EXPECT().
+		GetTree(gomock.Any()).
+		DoAndReturn(
+			func(_ string) ([]byte, kvstore.KVStore) {
+				return tree.Root(), nodeStore
+			}).
+		AnyTimes()
+
+	return treeStoreMock
+}
+
+func newTestP2PModule(t *testing.T, bus modules.Bus) modules.P2PModule {
+	t.Helper()
+
+	ctrl := gomock.NewController(t)
+	p2pMock := mockModules.NewMockP2PModule(ctrl)
+
+	p2pMock.EXPECT().Start().Return(nil).AnyTimes()
+	p2pMock.EXPECT().Stop().Return(nil).AnyTimes()
+	p2pMock.EXPECT().SetBus(gomock.Any()).Return().AnyTimes()
+	p2pMock.EXPECT().GetBus().Return(bus).AnyTimes()
+	p2pMock.EXPECT().
+		Broadcast(gomock.Any()).
+		Do(func(msg *anypb.Any) {
+		}).
+		AnyTimes()
+	p2pMock.EXPECT().
+		Send(gomock.Any(), gomock.Any()).
+		Do(func(addr cryptoPocket.Address, msg *anypb.Any) {
+		}).
+		AnyTimes()
+	p2pMock.EXPECT().GetModuleName().Return(modules.P2PModuleName).AnyTimes()
+	p2pMock.EXPECT().HandleEvent(gomock.Any()).Return(nil).AnyTimes()
+
+	return p2pMock
+}
+
+func newTestRuntimeConfig(t *testing.T) *runtime.Manager {
```

Comment:
Add a `// TECHDEBT` comment to centralize these types of helpers in the future.

cc @adshmh and @bryanchriswhite just for context & visibility